### ISSUE TITLE
Unexpected indentation level after newline with 0.50.0

### ISSUE THREAD

**Issue by rvandermeulen** [pinterest/ktlint#2103]
With version 0.50.0, the below code now produces an error because of the lack of newline after the {:
```
{ size -> icon?.getBitmap(size)?.await() }
```
What seems odd to me is that the "correct" formatting looks like the below:
```
{
        size ->
    icon?.getBitmap(size)?.await()
}
```
This plays out the same for larger code blocks as well where the first line after the bracket is indented further over than all subsequent ones if there's a separator involved. It seems to me that if anything, we'd want to invert that indentation to something more like:
```
{
    size ->
        icon?.getBitmap(size)?.await()
}
```

**Comment by paul-dingemans** [comment]
This problem has the same root cause as #2106 and will be fixed via that issue.

The double indentation of parameters inside the lambda looks indeed weird at first sight. It is based on (but not identical to) the default formatting of IntelliJ IDEA.

Given the functional literal below (note that it contains multiple parameters on separate lines):
```
val foo = { a,
        b,
        c ->
    doSomething()
}
```
It is formatted by IntelliJ IDEA formatter as:
```
val foo = { a,
            b,
            c ->
    doSomething()
}
```
Analog to formatting of functions having parameters on separate lines, ktlint formats this as:
```
val foo = {
        a,
        b,
        c ->
    doSomething()
}
```

The format above should only be applied when having a function literal with parameters on separate lines. As result of the bug in #2101, your parameter list was wrapped and as of that changed into a multiline parameter list. As a result of that, the indentation of a multiline parameter list was applied. With fix of #2106 you original code sample will be kept unchanged.

**Comment by rvandermeulen** [comment]
Thanks for the update, I appreciate the response!
